packages/embed/src/validation.ts

```diff
@@ -193,6 +206,60 @@ export const validateStore = ({
   return false
 }
 
+const buyerObject: Buyer = {
+  billingDetails: {
```

Comment:
Is there a reason we check billing details but not shipping details?

Reply:
Does this validation work with optional values? e.g. ` firstName=null`?

Reply:
Added validation for both and also for empty objects (like `billingDetails: {}` which is not valid)